sw/device/tests/crypto/cryptotest/firmware/ecdh.c

```diff
@@ -0,0 +1,92 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/crypto/impl/ecc/p256_common.h"
+#include "sw/device/lib/crypto/impl/integrity.h"
+#include "sw/device/lib/crypto/impl/keyblob.h"
+#include "sw/device/lib/crypto/include/datatypes.h"
+#include "sw/device/lib/crypto/include/ecc.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/lib/ujson/ujson.h"
+#include "sw/device/tests/crypto/cryptotest/json/ecdh_commands.h"
+
+status_t handle_ecdh_block(ujson_t *uj) {
+  // Declare ECDH parameter ujson deserializer types
+  cryptotest_ecdh_operation_t uj_op;
+  cryptotest_ecdh_curve_t uj_curve;
+  cryptotest_ecdh_private_key_t uj_private_key;
+  cryptotest_ecdh_coordinate_t uj_qx;
+  cryptotest_ecdh_coordinate_t uj_qy;
+
+  // Deserialize ujson byte stream into ECDH parameters
+  TRY(ujson_deserialize_cryptotest_ecdh_operation_t(uj, &uj_op));
+  TRY(ujson_deserialize_cryptotest_ecdh_curve_t(uj, &uj_curve));
+  TRY(ujson_deserialize_cryptotest_ecdh_private_key_t(uj, &uj_private_key));
+  TRY(ujson_deserialize_cryptotest_ecdh_coordinate_t(uj, &uj_qx));
+  TRY(ujson_deserialize_cryptotest_ecdh_coordinate_t(uj, &uj_qy));
+
+  otcrypto_ecc_curve_type_t curve_type;
+  otcrypto_unblinded_key_t public_key;
+  p256_point_t pub_p256;
+
+  switch (uj_curve) {
+    case kCryptotestEcdhCurveP256:
+      curve_type = kOtcryptoEccCurveTypeNistP256;
+      memset(pub_p256.x, 0, kP256CoordWords * 4);
+      memcpy(pub_p256.x, uj_qx.coordinate, uj_qx.coordinate_len);
+      memset(pub_p256.y, 0, kP256CoordWords * 4);
+      memcpy(pub_p256.y, uj_qy.coordinate, uj_qy.coordinate_len);
+      public_key.key_mode = kOtcryptoKeyModeEcdh;
+      public_key.key_length = sizeof(p256_point_t);
+      public_key.key = (uint32_t *)&pub_p256;
+      break;
+    default:
+      LOG_ERROR("Unsupported ECC curve: %d", uj_curve);
+      return INVALID_ARGUMENT();
+  }
+  public_key.checksum = integrity_unblinded_checksum(&public_key);
```

Comment:
Linking to https://github.com/lowRISC/opentitan/issues/20762 -- once there's an `import` function for ECDSA/ECDH keys, it would probably simplify this part. (Not a comment that should block this PR!)